src/Symfony/Component/Yaml/Parser.php

```diff
@@ -92,6 +92,18 @@ public function parse($value, $flags = 0)
             @trigger_error('Using the Yaml::PARSE_KEYS_AS_STRINGS flag is deprecated since version 3.4 as it will be removed in 4.0. Quote your keys when they are evaluable instead.', E_USER_DEPRECATED);
         }
 
+        if (Yaml::PARSE_FILE & $flags) {
+            if (!is_file($value)) {
+                throw new ParseException(sprintf('File "%s" does not exist.', $value));
+            }
+
+            if (!is_readable($value)) {
+                throw new ParseException(sprintf('File "%s" cannot be read.', $value));
+            }
+
+            $value = file_get_contents($value);
```

Comment:
In this case, we should ensure that `ParseException` have the file set in them when throwing them (either through the constructor or through `setParsedFile`). Otherwise, we don't benefit from the better error reporting.